### DISCUSSION TITLE
Trying to parse a string json data with packed event, getting error.

### DISCUSSION THREAD

**Discussion by Code-Crash** [101arrowz/fflate#143] [General]
I'm trying to parse the string JSON with the compress event as below, but it's throwing an error,

```js
let data = `{
  "token": "xyz",
  "event": "x%K\u000e \u0010ïò¯YH[LKô\u0000\u000b¢°P3\u001a\ráî¢n¿G\u001cÉÞ\n\u0013\u0007Êà¸.}EZ!E\u000e\u0012âÌ Æ\u000eúh£pT×Z0ÜÉÿÆ½Ò°¸)]¡±Í·µÖWm7\u00106r\u0002l#T"
}`
const event: any = JSON.parse(strFromU8(unzlibSync(strToU8(ll, true)))); // throws error
```

it's throwing `Error: invalid zlib data`

Is there any way to parse it somehow?

I want to parse the string first and uncompress the event, trying a bunch of things but not getting anywhere, please have a look.

Note: Actual packed data is being unencoded while posting the issue by github.

**Comment by 101arrowz** [comment]
You probably need to parse the JSON first.

```js
let data = `...`
const event = JSON.parse(data);
event.event = strFromU8(unzlibSync(strToU8(event.event, true)));
```

**Reply by Code-Crash** [reply]
@101arrowz When I'm trying to parse JSON, it is throwing an error, with `invalid control character in JSON string`

**Reply by 101arrowz** [reply]
How did you generate that string then? If you created it with some custom code (e.g. `const mystr = `{ "token": "xyz", event: "${myBinaryString}" }`) then it's not actually real JSON, you can't use it. You'll have to find the starting and ending quote manually, at which point you should reconsider your format.

**Reply by Code-Crash** [reply]
Thanks @101arrowz for your reply, actually JSON was valid, but I was reading this data from the file and I was converting first into `ASCII` which cause the issue, now I'm reading content directly which is working, JSON.parse is able to parse it.

Thanks.
